pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -214,16 +245,16 @@ func endpointsIPFamily(endpoints *v1.Endpoints) (*v1.IPFamily, error) {
 		return nil, fmt.Errorf("failed to identify ipfamily for endpoints (no addresses)")
 	}
 	// same apply on addresses
-	address := subset.Addresses[0]
-	if len(address.IP) == 0 {
+	ip := net.ParseIP(subset.Addresses[0].IP)
+	if ip == nil {
 		return nil, fmt.Errorf("failed to identify ipfamily for endpoints (address has no ip)")
 	}
 
 	ipv4 := v1.IPv4Protocol
 	ipv6 := v1.IPv6Protocol
-	if utilnet.IsIPv6String(address.IP) {
```

Comment:
The problem with this approach is that this does not handle the case with invalid IPs because we are assuming: "if is not IPv6 then it has to be IPv4". 
We had a similar bug with the legacy endpoint controller and headless services, where ClusterIP=None, 
don't know if this is possible with the endpoints Address.IP field, maybe it was already validated?

Reply:
endpoint IPs will always be actual IPs, not `"None"`